server/gokbg3/grails-app/services/org/gokb/FTUpdateService.groovy

```diff
@@ -438,10 +440,12 @@ class FTUpdateService {
           kbc.title?.historyEvents?.each{ he ->
             if (he.date){
               def event = [:]
-              event.date = he.date
-              event.from = he.from ?: ""
-              event.to = he.to ?: ""
-              event.uuid = he.uuid ?: ""
+              event.date = dateFormatService.formatIsoTimestamp(he.date)
+              event.from = []
+              event.from.addAll(he.from)
+              event.to = []
+              event.to.addAll(he.to)
+              event.id = he.id ?: ""
               result.titleHistory.add(event)
             }
           }
```

Comment:
`title.historyEvents` should be replaced with `title.titleHistory`. Simply doing `addAll` is not not enough here, since `from` & `to` are filled with TitleInstance objects, which cannot be automatically serialized in JSON. Therefore, every entry in `from`/`to` has to manually converted. Something like `he.from.collect { fe -> [id: fe.id, uuid: fe.uuid, name: fe.name ]}` would probably the shortest solution.

Reply:
Thanks for explaining again! Commit is going to follow.